vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -238,15 +238,30 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
         });
         grid._createPropertyObserver('activeItem', '__activeItemChangedDetails', true);
 
-        grid.$connector._getPageIfSameLevel = tryCatchWrapper(function (parentKey, index, defaultPage) {
-          let cacheAndIndex = grid._cache.getCacheAndIndex(index);
-          let parentItem = cacheAndIndex.cache.parentItem;
-          let parentKeyOfIndex = parentItem ? grid.getItemId(parentItem) : root;
-          if (parentKey !== parentKeyOfIndex) {
-            return defaultPage;
-          } else {
+        grid.$connector._getSameLevelPage = tryCatchWrapper(function (parentKey, index, defaultPage) {
```

Comment:
I checked and the current tests pass even without this change. I also tested the feature manually on the test page but didn't catch any infinite loops.

> Takes the parent items of the visible items into account when fetching a page. Tries to match the page number of the visible items, even if their parent is not visible but has the same depth as of the request. Previously, the the page number for the visible item was only considered if they were of the same depth as the request. This change prevents clearing the cache for the visible items, which can create a loop in some cases.

Do you have a scenario where the loop could be reproduced?

Reply:
I used a grid
- of depth 3
- with default page size of 50
- 200 root items
- 100 children per root item
- 50 grandchildren per child

It should be consistently reproducable when you call `scrollToIndex` for indexes "50-15-10".